mlir/include/mlir/Interfaces/ViewLikeInterface.td

```diff
@@ -414,4 +414,44 @@ def OffsetSizeAndStrideOpInterface : OpInterface<"OffsetSizeAndStrideOpInterface
   }];
 }
 
+def DistinctObjectsInterface : OpInterface<"DistinctObjectsInterface"> {
+  let description = [{
+    This intefaces indicates that pointer-like objects (such as memrefs) returned
```

Comment:
Can we phrase the interface such that it 's not limited to `memref`? It sounds like this entire mechanism - and, to my knowledge, the alias analysis being modified - isn't actually limited to `memref` and could work on things like `ptr.ptr`.

Reply:
Wait, I specifically tried to phrase it `pointer-like objects` so `ptr.ptr` or similar should qualify. I can remove `(such as memref)` if you think it will be better.